platform/linux-generic/include/odp/api/plat/atomic_inlines.h

```diff
@@ -264,6 +296,50 @@ _ODP_INLINE uint64_t odp_atomic_xchg_u64(odp_atomic_u64_t *atom,
 	return ATOMIC_OP(atom, atom->v = new_val);
 }
 
+_ODP_INLINE void odp_atomic_bit_set_u64(odp_atomic_u64_t *atom, uint64_t bits)
+{
+	uint64_t old_val;
+
+	old_val = odp_atomic_load_u64(atom);
+
+	while (!odp_atomic_cas_u64(atom, &old_val, old_val | bits))
```

Comment:
These functions could be implemented using the ATOMIC_OP macro similarly to the adjacent functions. Now both the atomic load and the cas expand to a spinlocked section. So instead of taking the spinlock once, this now takes it two or more times.

Reply:
V2.